misc_docs/syntax/decorator_uncurry.mdx

```diff
@@ -0,0 +1,34 @@
+---
+id: "uncurry-decorator"
+keywords: ["uncurry", "decorator"]
+name: "@uncurry"
+summary: "This is the `@uncurry` decorator."
+category: "decorators"
+---
+
+The `@uncurry` decorator allows you to annotate **external** function callback argument types
+so when writing the callbacks they may be declared as normal functions `() => { ... }`
+rather than needing the uncurried function syntax `(.) => { ... }`.
+
+### Example
+
+<CodeTab labels={["ReScript", "JS Output"]}>
+
+```res
+@bs.send
+external map: (array<'a>, @uncurry ('a => 'b)) => array<'b> = "map"
+
+let result = map([1, 2, 3], x => x + 1)
+```
+
+```js
+var result = [1, 2, 3].map(function (x) {
+  return (x + 1) | 0
+})
+```
+
+</CodeTab>
+
+### References
+
+- [Curry & Uncurry](/docs/manual/latest/bind-to-js-function#curry--uncurry)
```

Comment:
Super important section to include as a reference:
https://rescript-lang.org/docs/manual/latest/bind-to-js-function#solution-use-guaranteed-uncurrying

Reply:
👍 updated